core/network/corda-interop-app/interop-workflows/src/main/kotlin/com/weaver/corda/app/interop/flows/AssetTransferWrapperFlows.kt

```diff
@@ -0,0 +1,289 @@
+import com.weaver.corda.app.interop.states.AssetPledgeState
+import com.weaver.corda.app.interop.states.AssetPledgeParameters
+import com.weaver.corda.app.interop.states.AssetReclaimParameters
+import com.weaver.corda.app.interop.states.AssetClaimParameters
+
+import net.corda.core.contracts.ContractState
+import net.corda.core.contracts.CommandData
+import net.corda.core.contracts.Command
+import net.corda.core.contracts.UniqueIdentifier
+import net.corda.core.contracts.TimeWindow
+import net.corda.core.contracts.StateAndRef
+import net.corda.core.contracts.requireThat
+import net.corda.core.contracts.StaticPointer
+
+import net.corda.core.identity.Party
+import net.corda.core.identity.CordaX500Name
+
+import net.corda.core.flows.*
+import net.corda.core.node.services.queryBy
+import net.corda.core.node.services.vault.QueryCriteria
+import net.corda.core.node.ServiceHub
+import net.corda.core.transactions.TransactionBuilder
+import net.corda.core.transactions.SignedTransaction
+import net.corda.core.utilities.OpaqueBytes
+import java.time.Instant
+import java.util.Base64
+
+import com.weaver.protos.common.asset_locks.AssetLocks
+import com.weaver.protos.common.asset_transfer.AssetTransfer
+
+/**
+ * The AssetTransferPledgeFungible flow is used to create a pledge for a fungible asset.
+ *
+ * @property pledgeArgs The details of the fungible asset being pledged for transfer to remote network.
+ *                          Bleow properties are captured by this structure.
+ * assetType -- The type of the fungible asset to be pledged.
+ * assetIdOrQuantity -- The number of units of the fungible asset (e.g., tokens) to be pledged.
+ * recipientCert -- The certificate of the party in the remote network who will be the asset owner after transfer.
+ * expiryTimeSecs -- The time in epoch till which the pledge is valid.
+ * getAssetStateAndRefFlow -- The name of the flow used to fetch the fungible asset's StateAndRef from the vault.
+ * deleteAssetStateCommand -- The name of the command used to delete the pledged fungible asset.
+ * issuer -- The issuing authority of the pledged fungible asset.
+ * observers -- The parties who are not transaction participants but only observers.
+ */
+@InitiatingFlow
+@StartableByRPC
+class AssetTransferPledgeFungible
+constructor(
+    val pledgeArgs: AssetPledgeParameters
+) : FlowLogic<Either<Error, UniqueIdentifier>>() {
+    /**
+     * The call() method captures the logic to create a new [AssetPledgeState] state in the vault.
+     *
+     * It first creates a new AssetPledgeState. It then builds
+     * and verifies the transaction, collects the required signatures,
+     * and stores the state in the vault.
+     *
+     * @return Returns the contractId of the newly created [AssetPledgeState].
+     */
+    @Suspendable
+    override fun call(): Either<Error, UniqueIdentifier> = try {   
+
+        val numUnits: Long = pledgeArgs.assetIdOrQuantity.toLong()
```

Comment:
I thought the application flow function was doing this conversion? and I feel if we are letting application flow decide whether `assetIdOrQuantity` is Id or Quantity, then we can even avoid the two Pledge Functions, and have only one, right? Because that's the only thing that separates the two functions.

Reply:
As discussed, now using the data type `Any` to accommodate both the `assetId (String)` for non-fungible and `quantity (Long)` for fungible assets.